lib/cosmos/dart/lib/dart_common.rb

```diff
@@ -0,0 +1,471 @@
+require File.expand_path('../../config/environment', __FILE__)
+require 'cosmos/script'
+require 'optparse'
+
+# Autoload models here to remove problems loading within Cosmos namespace
+Target
+Packet
+PacketLog
+PacketLogEntry
+
+module DartCommon
+
+  MAX_STRING_BYTE_SIZE = 191 # Works well with mysql utf8mb4 if we want to support mysql in the future
+  MAX_STRING_BIT_SIZE = MAX_STRING_BYTE_SIZE * 8
+  MAX_COLUMNS_PER_TABLE = 200
+  REDUCED_TYPES = [:integer, :bigint, :decimal, :float]
+
+  def self.handle_argv
+    parser = OptionParser.new do |option_parser|
+      option_parser.banner = "Usage: ruby #{option_parser.program_name} [options]"
+      option_parser.separator("")
+    
+      # Create the help option
+      option_parser.on("-h", "--help", "Show this message") do
+        puts option_parser
+        exit
+      end
+    
+      # Create the version option
+      option_parser.on("-v", "--version", "Show version") do
+        puts "COSMOS Version: #{COSMOS_VERSION}"
+        puts "User Version: #{USER_VERSION}" if defined? USER_VERSION
+        exit
+      end
+    
+      # Create the system option
+      option_parser.on("--system FILE", "Use an alternative system.txt file") do |arg|
+        System.instance(File.join(USERPATH, 'config', 'system', arg))
+      end
+    end.parse!
+  end
+
+  def get_decom_table_model(packet_config_id, table_index, reduction_modifier = "")
+    model_name = "T#{packet_config_id}_#{table_index}#{reduction_modifier}"
+    begin
+      model = Cosmos.const_get(model_name)
+    rescue
+      # Need to create model
+      model = Class.new(ActiveRecord::Base) do
+        self.table_name = "t#{packet_config_id}_#{table_index}#{reduction_modifier}"
+      end
+      Cosmos.const_set(model_name, model)
+    end
+    return model
+  end
+
+  def cosmos_data_type_to_db_type(data_type, bit_size, array_size)
+    db_type = nil
+    case data_type
+    when :INT
+      if bit_size <= 32
+        db_type = :integer
+      elsif bit_size <= 64
+        db_type = :bigint
+      else
+        db_type = :decimal
+      end
+    when :UINT
+      if bit_size <= 31
+        db_type = :integer
+      elsif bit_size <= 63
```

Comment:
Because the SQL database doesn't have an unsigned integer type.  So a 32bit unsigned number would not store correctly into the database type (integer or bigint)